### PR TITLE
Migration from Dockerfile-based builds to bazel

### PR DESCRIPTION
**Pull request by geojaz**
Migrating from dockerfile based builds to bazel. This is WIP. Will tag when we're ready to review and I'll try to limit the scope/size of this to make it reasonable to review.

Changelog:
* revised .gitignore for bazel workflow
* updated stackdriver exporter imports because the library moved into contrib
* building binaries in bazel
* building images in bazel

To come for this PR:
- [x] verify that these images work by swapping these out with the existing 
- [x] update Makefile for this bazel workflow

There is a bunch of follow-on work to come but given the size of this PR, I'd like to review and get this one done first as it can merge without them.

### CODE REVIEW

File: pyrios/main.go [hunk 1]
```diff
@@ -26,7 +26,8 @@ import (
 	"cloud.google.com/go/compute/metadata"
 	"context"
 	"fmt"
-	"go.opencensus.io/exporter/stackdriver"
+	// "go.opencensus.io/exporter/stackdriver"
```
Comment (chrislovecnm) [inline pyrios/main.go]: lets cleanup comments

Comment (geojaz) [reply pyrios/main.go]: Lots of changes coming ;)

**Comment by geojaz** [comment]
git issues to fix... however, the content is essentially as I'd like reviewed so please feel free to take a look. the images that were built here are gcr.io/pso-examples/pyrios:latest and gcr.io/pso-examples/pyrios-ui:latest

**Comment by geojaz** [comment]
OK git issues are wrapped up and i pulled the fixes for pyrios out to a separate PR since it is needed regardless of bazel support

File: BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2018 Google LLC
```
Comment (chrislovecnm) [inline BUILD.bazel]: We probably want to add some documentation to the bazel files :) Major docs

File: Makefile [hunk 3]
```diff
@@ -118,3 +135,87 @@ expose-ui:
...
+		--platforms=@io_bazel_rules_go//go/toolchain:linux_amd64 \
+		--define REGISTRY=${IMAGE_REGISTRY} \
+		--define REPOSITORY=${PYRIOS_UI_REPO} \
+		--define TAG=${PYRIOS_UI_TAG} \
+		//pyrios-ui:go_image
+
+.PHONY: bazel-build-images
+bazel-build-images: bazel-build-pyrios-image bazel-build-pyrios-ui-image
+
+.PHONY: bazel-push-pyrios
```
Comment (chrislovecnm) [inline Makefile]: Need docs all through here

File: pyrios-ui/BUILD.bazel [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+go_image(
+    name = "go_image_base",
+    embed = [":go_default_library"],
+    goarch = "amd64",
+    goos = "linux",
+    pure = "on",
+)
+
+container_image(
```
Comment (chrislovecnm) [inline pyrios-ui/BUILD.bazel]: What image does this use? We probably want to use a google maintained image

File: pyrios-ui/manifests/deployment.yaml [hunk 5]
```diff
@@ -32,10 +32,10 @@ spec:
     spec:
       containers:
       - name: pyrios-ui
-        image: gcr.io/pso-examples/pyrios-ui:1.0.1
+        image: gcr.io/pso-examples/pyrios-ui:latest
```
Comment (chrislovecnm) [inline pyrios-ui/manifests/deployment.yaml]: yah no latest please.  Not sure how to version easily, but we need to version

File: pyrios/main.go [hunk 6]
```diff
@@ -26,7 +26,7 @@ import (
 	"cloud.google.com/go/compute/metadata"
```
Comment (chrislovecnm) [inline pyrios/main.go]: can you format these.  We need the formatter we use in kops

File: pyrios/manifests/deployment.yaml [hunk 7]
```diff
@@ -32,7 +32,7 @@ spec:
     spec:
       containers:
       - name: pyrios
-        image: gcr.io/pso-examples/pyrios:1.0.0
+        image: gcr.io/pso-examples/pyrios:latest
```
Comment (chrislovecnm) [inline pyrios/manifests/deployment.yaml]: Again no latest please

File: pyrios-ui/BUILD.bazel (hunk 4 above)
Comment (geojaz) [reply pyrios-ui/BUILD.bazel]: We are using google's distroless images: https://github.com/GoogleContainerTools/distroless/blob/master/examples/go/Dockerfile

File: pyrios/main.go (hunk 6 above)
Comment (geojaz) [reply pyrios/main.go]: my bad, when testing is hooked up, we'll catch that :)

File: pyrios-ui/manifests/deployment.yaml (hunk 5 above)
Comment (geojaz) [reply pyrios-ui/manifests/deployment.yaml]: I have a plan for versioning and will get it into next iteration!

File: BUILD.bazel (hunk 2 above)
Comment (geojaz) [reply BUILD.bazel]: I will provide full documentation on both repo/demo flow/development flow and also walkthrough of setting up bazel like this. 

But, if it's ok, I'd like to finish moving things around first and get the tests going ( issues #3 and #4 ). I think taking care of those issues will help to focus the docs.

File: pyrios-ui/BUILD.bazel (hunk 4 above)
Comment (chrislovecnm) [reply pyrios-ui/BUILD.bazel]: perfect

**Pull request merged**